compiler/rustc_passes/src/dead.rs

```diff
@@ -379,6 +380,17 @@ impl<'tcx> MarkSymbolVisitor<'tcx> {
                 _ => intravisit::walk_item(self, item),
             },
             Node::TraitItem(trait_item) => {
+                // FIXME: could we get all impl terms for the trait item?
+                // then we can add them into worklist when the trait item is live,
+                // so that we won't lose any lost chain in the impl terms.
+                // this also allows us to lint such code:
+                // ```rust
+                // struct UnusedStruct; //~ WARN
+                // trait UnusedTrait { //~ WARN
+                //     fn foo() {}
+                // }
+                // impl UnusedTrait for UnusedStruct { fn foo() {} }
```

Comment:
https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/context/struct.TyCtxt.html#method.for_each_impl

Reply:
I want some methods to give all ImpIterm for a given TrairTerm, for example, get corresponding method foo in impl Trait for T if given the DefId of Trait::foo.
Maybe my words make the misunderstood because my English is poor.
Are there such methods? Hopefully I make it clearer.
And I found this method previous, but we need to traverse all TraitItems to find expected one and traverse them again for another one.